powerstore/resource_volume.go

```diff
@@ -3,261 +3,334 @@ package powerstore
 				Optional:            true,
-				Computed:            true,
 				Description:         "The host name of the volume.",
 				MarkdownDescription: "The host name of the volume.",
+				Validators: []validator.String{
+					stringvalidator.ConflictsWith(path.MatchRoot("host_id")),
+				},
 			},
-			"host_group_id": {
-				Type:                types.StringType,
+
+			"host_group_id": schema.StringAttribute{
 				Optional:            true,
 				Computed:            true,
 				Description:         "The host group id of the volume.",
 				MarkdownDescription: "The host group id of the volume.",
 			},
-			"host_group_name": {
-				Type:                types.StringType,
+			"host_group_name": schema.StringAttribute{
 				Optional:            true,
-				Computed:            true,
 				Description:         "The host group name of the volume.",
 				MarkdownDescription: "The host group name of the volume.",
+				Validators: []validator.String{
+					stringvalidator.ConflictsWith(path.MatchRoot("host_group_id")),
+				},
 			},
-			"logical_unit_number": {
-				Type:                types.Int64Type,
+			"logical_unit_number": schema.Int64Attribute{
 				Computed:            true,
 				Optional:            true,
 				Description:         "The current amount of data written to the volume.",
 				MarkdownDescription: "The current amount of data written to the volume.",
 			},
-			"volume_group_id": {
-				Type:                types.StringType,
+			"volume_group_id": schema.StringAttribute{
 				Computed:            true,
 				Optional:            true,
 				Description:         "The volume group id of the volume.",
 				MarkdownDescription: "The volume group id of the volume.",
 			},
-			"volume_group_name": {
-				Type:                types.StringType,
-				Computed:            true,
+			"volume_group_name": schema.StringAttribute{
 				Optional:            true,
 				Description:         "The volume group name of the volume.",
 				MarkdownDescription: "The volume group name of the volume.",
+				Validators: []validator.String{
+					stringvalidator.ConflictsWith(path.MatchRoot("volume_group_id")),
+				},
 			},
-			"min_size": {
-				Type:                types.Int64Type,
-				Computed:            true,
+			"min_size": schema.Int64Attribute{
 				Optional:            true,
 				Description:         "The minimum size  of the volume.",
 				MarkdownDescription: "The minimum size of the volume.",
 			},
-			"sector_size": {
-				Type:                types.Int64Type,
-				Computed:            true,
+			"sector_size": schema.Int64Attribute{
 				Optional:            true,
+				Computed:            true,
 				Description:         "The sector size of the volume.",
 				MarkdownDescription: "The sector size of the volume.",
-				PlanModifiers: tfsdk.AttributePlanModifiers{
-					DefaultAttribute(types.Int64{Value: defaultSectorSize}),
+				PlanModifiers: []planmodifier.Int64{
+					DefaultAttribute(int64(defaultSectorSize)),
```

Comment:
the sector size field is present only at the time of creation, and if absent the API gives error which is not understandable by user. It can be modified in the main.tf file only if absent will have default value.

Reply:
My point is that, these defaults should not be defined in this repo. It can either be defined in gopowerstore repo, or this field should be required. 
If the API gives an error if this field is not set, I would say that the provider should exhibit the same behaviour.